app/vmctl/vm_native.go

```diff
@@ -109,6 +112,12 @@ func (p *vmNativeProcessor) do(ctx context.Context, f native.Filter, srcURL, dst
 
 func (p *vmNativeProcessor) runSingle(ctx context.Context, f native.Filter, srcURL, dstURL string) error {
 
+	matchWithFilters, err := buildMatchWithFilter(p.filter.Match, f.Match)
```

Comment:
The `f.Match` field name suggests that it can contain arbitrary metricsql filters. But the `buildMatchWithFilter()` function contents suggests that the second arg should be a metric name. Probably, it is time to rename `f.Match` to `f.MetricName` in order to make the code less misleading?

Reply:
I have moved the function to the correct place. This helped to remove the misleading understanding of the arguments and avoid the struct changes.